### PR TITLE
Add dtype choice in step type/functions

### PR DESCRIPTION
**Pull request by thomashirtz**
See #256 

I just feel that it is a little weird that this dtype argument is both responsible for both reward and discount in "restart". But otherwise it means that there are no way to choose reward's dtype at restart time.

### CODE REVIEW

File: jumanji/types.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def restart(
     observation: Observation,
     extras: Optional[Dict] = None,
     shape: Union[int, Sequence[int]] = (),
+    dtype: Union[jnp.dtype, type] = float,
```
Comment (sash-a) [inline jumanji/types.py:98]: Can you please add this to the docs for all these methods :pray:

**Review by sash-a — CHANGES_REQUESTED** [review]
Thanks for this @thomashirtz 

Can you please add the dtype arg to the docstring and also sign the CLA :smile:

File: jumanji/types.py (hunk 1 above)
Comment (thomashirtz) [reply jumanji/types.py:98]: Oh mb, I forgot about it, it's done, tell me if it is okay :)

**Review by sash-a — APPROVED** [review]
Thanks @thomashirtz. Can you please sign the contributor license agreement, click the link in the first comment :pray:
